roles/gpu_operator_bundle_from_commit/files/operator_bundle_image_builder_script.sh

```diff
@@ -0,0 +1,206 @@
+#! /bin/bash
+
+set -o pipefail
+set -o errexit
+set -o nounset
+set -x
+
+LOCAL_OPERATOR_IMAGE_REPOSITORY="image-registry.openshift-image-registry.svc:5000/gpu-operator-ci"
+LOCAL_OPERATOR_IMAGE_NAME="gpu-operator-ci"
+
+echo "QUAY_BUNDLE_IMAGE_NAME=${QUAY_BUNDLE_IMAGE_NAME}"
+
+if [ "${PUBLISH_TO_QUAY:-}" ]; then
+    if [ "${WITH_DRIVER:-}" ]; then
+        echo "FATAL: Cannot have PUBLISH_TO_QUAY and WITH_DRIVER enabled together."
+        exit 1
+    fi
+
+    OPERATOR_IMAGE_REPOSITORY="$(dirname "$QUAY_BUNDLE_IMAGE_NAME")"
+    OPERATOR_IMAGE_NAME="$(basename "$QUAY_BUNDLE_IMAGE_NAME")"
+else
+    OPERATOR_IMAGE_REPOSITORY="${LOCAL_OPERATOR_IMAGE_REPOSITORY}"
+    OPERATOR_IMAGE_NAME="${LOCAL_OPERATOR_IMAGE_NAME}"
+fi
+
+BUNDLE_IMAGE_FULL="${QUAY_BUNDLE_IMAGE_NAME}:operator_bundle_${OPERATOR_IMAGES_TAG_SUFFIX}"
+
+CONTAINER_FILE=./docker/bundle.Dockerfile
+
+CONTEXT_LOCATION="."
+
+CSV_FILE=bundle/manifests/gpu-operator.clusterserviceversion.yaml
+
+(echo "{ \"auths\": " ; cat /var/run/secrets/openshift.io/push/.dockercfg ; echo "}") > /tmp/.dockercfg_local
+LOCAL_AUTH="--tls-verify=false --authfile /tmp/.dockercfg_local"
+
+cat /var/run/secrets/quay.io/push/.dockerconfigjson > /tmp/.dockercfg_quay
+QUAY_AUTH="--tls-verify=false --authfile /tmp/.dockercfg_quay"
+
+# ---
+
+replace_related_image() {
+    NAME=$1
+    shift
+    NEW_IMAGE=$1
+
+    OLD_IMAGE=$(cat "${CSV_FILE}" | yq \
+                    | jq -r '.spec.relatedImages[] | select(.name | test("'$NAME'")) | .image')
```

Comment:
Should probably be `==` rather than `| test(`

Reply:
indeed (but this part part goes away with your next pure `yq` suggestion)